File: examples/widgets/arc/lv_example_arc_1.c
#include "../../lv_examples.h"

#if LV_USE_ARC && LV_BUILD_EXAMPLES

static void value_changed_event_cb(lv_event_t * e);

/**
 * @title Arc with rotating percentage label
 * @brief Report the arc value on a label that orbits with the knob.
 *
 * A 150 by 150 arc sweeps 270 degrees starting at rotation 135 and
 * begins at value 10. An `LV_EVENT_VALUE_CHANGED` callback writes the
 * current value into a label as `<v>%` and calls
 * `lv_arc_rotate_obj_to_angle` with a 25 px radius so the label
 * follows the knob. The initial label is primed with
 * `lv_obj_send_event`.
 */
void lv_example_arc_1(void)
{
    lv_obj_t * label = lv_label_create(lv_screen_active());

    /*Create an Arc*/
    lv_obj_t * arc = lv_arc_create(lv_screen_active());
    lv_obj_set_size(arc, 150, 150);
    lv_arc_set_rotation(arc, 135);
    lv_arc_set_bg_angles(arc, 0, 270);
    lv_arc_set_value(arc, 10);
    lv_obj_center(arc);
    lv_obj_add_event_cb(arc, value_changed_event_cb, LV_EVENT_VALUE_CHANGED, label);

    /*Manually update the label for the first time*/
    lv_obj_send_event(arc, LV_EVENT_VALUE_CHANGED, NULL);
}

static void value_changed_event_cb(lv_event_t * e)
{
    lv_obj_t * arc = lv_event_get_target_obj(e);
    lv_obj_t * label = (lv_obj_t *)lv_event_get_user_data(e);

    lv_label_set_text_fmt(label, "%" LV_PRId32 "%%", lv_arc_get_value(arc));

    /*Rotate the label to the current position of the arc*/
    lv_arc_rotate_obj_to_angle(arc, label, 25);
}

#endif

File: examples/widgets/arc/lv_example_arc_1.py
def value_changed_event_cb(e,label):
    arc = e.get_target_obj()

    txt = "{:d}%".format(arc.get_value())
    label.set_text(txt)

    # Rotate the label to the current position of the arc
    arc.rotate_obj_to_angle(label, 25)

label = lv.label(lv.scr_act())

# Create an Arc
arc = lv.arc(lv.scr_act())
arc.set_size(150, 150)
arc.set_rotation(135)
arc.set_bg_angles(0, 270)
arc.set_value(10)
arc.center()
arc.add_event(lambda e: value_changed_event_cb(e,label),lv.EVENT.VALUE_CHANGED, None)

# Manually update the label for the first time
arc.send_event(lv.EVENT.VALUE_CHANGED, None)

